aws/table_aws_securityhub_finding.go

```diff
@@ -412,9 +415,56 @@ func getSecurityHubFinding(ctx context.Context, d *plugin.QueryData, _ *plugin.H
 func buildListFindingsParam(quals plugin.KeyColumnQualMap) *types.AwsSecurityFindingFilters {
 	securityFindingsFilter := &types.AwsSecurityFindingFilters{}
 	strFilter := types.StringFilter{}
+	dateFilter := types.DateFilter{}
+	timeFormat := "2006-01-02T15:04:05Z"
 
 	strColumns := []string{"company_name", "compliance_status", "generator_id", "product_arn", "product_name", "record_state", "title", "verification_state", "workflow_state", "workflow_status", "source_account_id"}
 
+	timeColumns := []string{"created_at", "updated_at"}
+
+	for _, t := range timeColumns {
+		if quals[t] == nil {
+			continue
+		}
+		for _, q := range quals[t].Quals {
+			value := q.Value.GetTimestampValue().AsTime().Format(timeFormat)
+			if value == "" {
+				continue
+			}
+
+			switch q.Operator {
+			case "=", ">=", ">":
+				dateFilter.Start = &value
+				dateFilter.End = aws.String(time.Now().Format(timeFormat))
+			case "<", "<=":
+				dateFilter.End = &value
+				st, err := time.Parse(timeFormat, value)
+				if err != nil {
+					panic("failed to parsing provided value " + value + " for " + t)
+				}
+				if t == "updated_at" {
+					// Default to past 90 days.
+					// https://docs.aws.amazon.com/securityhub/latest/userguide/securityhub-findings.html
+					dateFilter.Start = aws.String(st.AddDate(0, 0, -90).Format(timeFormat))
+				} else {
+					// For the query "select * from aws_securityhub_finding where created_at <= current_timestamp - interval '31d'", we are setting the end time based on the query parameter.
+					// The query doesn't explicitly mention a start date for the creation time.
+					// AWS retains Security Hub findings updated within the last 90 days, regardless of when they were created.
+					// There is no strict limit for the creation start time, but we have set it to the last five years. The API will return an error if we will not set the Start anf End time all together.
+					// Based on our testing, a five-year window seems to be an optimal period.
+					dateFilter.Start = aws.String(st.AddDate(-5, 0, 0).Format(timeFormat))
```

Comment:
@misraved, For the query `select * from aws_securityhub_finding where created_at <= '2024-08-07T23:47:22+05:30'`:

- We are setting the "Create End" time exactly as specified in the query parameter.
- For the "Create Start" time, we are defaulting it to five years prior in the background, which would be `'2019-08-07T23:47:22+05:30'`.

Note: Both the start and end times must be set together. If one is provided, both need to be specified.

Does this behavior for the table seem appropriate?

Reply:
@ParthaI @misraved when we can expect this to be merged?

Reply:
Hi @michalz-rely, we will release the plugin with these changes on Monday.

Reply:
@ParthaI I believe you can set the "create start" time to `2018-11-27T00:00:00Z` (27 November 2018) which marks the date when AWS introduced `SecurityHub`.

It might be worth testing a time period prior to `2018-11-27T00:00:00Z` just to understand if the API returns any error or not.